src/v/redpanda/admin/server.cc

```diff
@@ -1860,6 +1864,21 @@ admin_server::patch_cluster_config_handler(
                       yaml_name,
                       property.format_raw(yaml_value),
                       validation_err.value().error_message());
+                } else if (auto restricted_err = property.check_restricted(val);
+                           restricted_err.has_value() && !valid_license) {
+                    // TODO(oren): prob better to format the whole message like
+                    // the log?
+                    errors[yaml_name] = restricted_err.value().error_message();
```

Comment:
I was thinking about whether we need to perform a similar check on the properties in `update.remove`. I think it's a fair assumption to make that the default value is not restricted, so we don't need to, but maybe we should make that assumption explicit with a vassert somewhere (property constructor?). What do you think?

Reply:
you mean like `assert(check_restricted(this->default_val()) == false`? yeah I'm working on something like that currently 🙂

Reply:
Nice

Reply:
done